pkg/plugin/twinmaker/utils.go

```diff
@@ -141,12 +141,43 @@ type PropertyReference struct {
 	entityName              *string
 }
 
-func GetEntityPropertyReferenceKey(entityPropertyReference *iottwinmaker.EntityPropertyReference) (s string) {
+/* 
+ * PropertyDefinitions input:
+ *
+ * "propertyDefinitions": {
+    "PROPERTY_NAME": {
+      "dataType": {
+        "type": DATA_TYPE
+      },
+      ...
+      "isExternalId": IS_EXTERNAL_ID,
+	  ...
+    },
+ * 
+ * IsExternalId map output:
+ * 
+ * {
+	 ...
+	 "PROPERTY_NAME": IS_EXTERNAL_ID,
+	 ...
+   }
+*/
+func ConvertPropertyDefinitionsToIsExternalIdMap(propertyDefinitions map[string]*iottwinmaker.PropertyDefinitionResponse) (map[string]bool) {
```

Comment:
Does this function check if the property has the external ID? If so maybe the name is misleading since we are not converting but verifying?

Reply:
We're not checking if a property is an externalId with this function. This is a utility function to build a mapping that will make checking that a property is an externalId faster.
I can simplify this and remove this function